src/components/VideoQueueItem.vue

```diff
@@ -50,8 +50,23 @@ export default {
 			isLoadingVote: false,
 			hasBeenAdded: false,
 			thumbnailHasError: false,
+			existsInQueue: false,
 		};
-	},
+	}, 
+	created() { 
+        if (this.item.id === this.$store.state.room.currentSource.id) {
+			this.isLoadingAdd = false;
+			this.hasBeenAdded = true;
+			this.existsInQueue = true;
+		}
+		for (let i = 0; i < this.$store.state.room.queue.length; i++) {
```

Comment:
Use a `for...of` loop here instead

Reply:
The loop should also return early

Reply:
I added a break, lmk if I didn't do this correctly though